app/models/user.rb

```diff
@@ -1,2 +1,6 @@
 class User < ApplicationRecord
+  def user_tests(level)
+    test_ids = Result.where(user_id: id).select(:test_id)
+    Test.where(id: test_ids).where(level: level)
```

Comment:
два `where` можно объединить
```suggestion
    test_ids = Result.where(user_id: id).select(:test_id)
    Test.where(id: test_ids, level: level)
```
и посмотри комментарий в начале ревью - здесь стоит реализовывать с использованием `joins`